core/kernel/notif_default.c

```diff
@@ -1,97 +1,169 @@
 // SPDX-License-Identifier: BSD-2-Clause
 /*
- * Copyright (c) 2021-2023, Linaro Limited
+ * Copyright (c) 2021-2024, Linaro Limited
  */
 
 #include <assert.h>
 #include <bitstring.h>
+#include <config.h>
+#include <initcall.h>
 #include <kernel/interrupt.h>
 #include <kernel/notif.h>
 #include <kernel/spinlock.h>
+#include <kernel/virtualization.h>
 #include <trace.h>
 #include <types_ext.h>
 
-static bitstr_t bit_decl(notif_values, NOTIF_ASYNC_VALUE_MAX + 1);
-static bitstr_t bit_decl(notif_alloc_values, NOTIF_ASYNC_VALUE_MAX + 1);
+struct notif_vm_bitmap {
+	bool alloc_values_inited;
+
```

Comment:
could remove this empty line.

Reply:
Sure. There was a warning in checkpatch that was complaining about there not being an empty line.

Reply:
>> checkdiff 383d05944b81ab269736348b7e533ba12bbe2f03 37e5be45ebdf90c2ec22ea144e88155969fd6cd2
WARNING: Missing a blank line after declarations
#60: FILE: core/kernel/notif_default.c:19:
+	bool alloc_values_inited;
+	bitstr_t bit_decl(values, NOTIF_ASYNC_VALUE_MAX + 1);